core/Tracker/Visit/VisitProperties.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Tracker\Visit;
+
+/**
+ * Holds temporary data for tracking requests.
+ *
+ * RequestProcessors
+ */
+class VisitProperties
+{
+    /**
+     * Information about the current visit. This array holds the column values that will be inserted or updated
+     * in the `log_visit` table, or the values for the last known visit of the current visitor.
+     *
+     * @var array
+     */
+    public $visitorInfo = array();
+
+    /**
+     * Stores plugin specific tracking request metadata. RequestProcessors can store
+     * whatever they want in this array, and other RequestProcessors can modify these
+     * values to change tracker behavior.
+     *
+     * @var string[][]
+     */
+    private $requestMetadata = array();
+
+    /**
+     * Set a request metadata value.
+     *
+     * @param string $pluginName eg, `'Actions'`, `'Goals'`, `'YourPlugin'`
+     * @param string $key
+     * @param mixed $value
+     */
+    public function setRequestMetadata($pluginName, $key, $value)
+    {
+        $this->requestMetadata[$pluginName][$key] = $value;
```

Comment:
I find this concept a bit "weird". Would it make sense to `Request::setMetadata()` on the request object? Visitor and request metadata doesn't sound like it belongs together maybe? Also maybe we can remove one dimension instead of `$pluginName` and `$key` just have `$name` if possible? Plugin name would be acceptable for me only if it was set by the platform automatically, if possible. 

Is it possible to not have metadata but clear objects and what can be done on the `Visitor` object? Eg `$visitor->setIsVisitorKnown($isKnown)`? One problem that I have is that eg the core knows about plugins. Eg within core (`/core`) we do `getRequestMetadata('CoreHome', 'isVisitorKnown')`, but core should not know anything about the plugins.